dev_guide/templates.adoc

```diff
@@ -239,9 +239,159 @@ application at will.
 
 [NOTE]
 ====
-Some of the QuickStart templates define a database
+Some of the InstantApp templates define a database
 link:../architecture/core_concepts/deployments.html#deployments-and-deployment-configurations[deployment
 configuration]. The configuration they define uses ephemeral storage for the
 database content. These templates should be used for demonstration purposes only
 as all database data will be lost if the database pod restarts for any reason.
 ====
+
+[[writing-templates]]
+== Writing Templates
+Developers can define new templates to make it easy to recreate all the resources of their application.  The template will define the resources it creates along with some metadata to guide the creation of those resources.
```

Comment:
yeah, i debated...  Jorge's blog is in theory going to get ported into the docs by the docs team and it talks more about that.  but i guess i'll go ahead and mention it since i'm in here anyway.